### PR TITLE
add checks on inputs

### PR DESCRIPTION
**Pull request by Krovatkin**

### CODE REVIEW

File: torch/csrc/jit/tensorexpr/kernel.cpp [hunk 1]
```diff
@@ -954,8 +955,85 @@ void TensorExprKernel::LowerToBackend(BackendType backend_type) {
...
+    if (!isValidPrimProperty(vs[i], sz[i])) {
+      return false;
+    }
+  }
+  return true;
+}
+
 void TensorExprKernel::PickAndCheckBackendType(
     const at::ArrayRef<IValue>& inputs) {
+  TORCH_INTERNAL_ASSERT(
```
Comment (bertmaher) [inline torch/csrc/jit/tensorexpr/kernel.cpp:1071]: Could you possibly pull this out into a helper function so it doesn't impede the readability of the "main" path?

**Review by bertmaher — APPROVED** [review]
Looks good, thanks for doing this!  Just one comment inline.

**Comment by Krovatkin** [comment]
Will do!!!

File: torch/csrc/jit/tensorexpr/kernel.cpp (hunk 1 above)
Comment (Krovatkin) [reply torch/csrc/jit/tensorexpr/kernel.cpp:1071]: Will do!

**Pull request merged**